python/lsst/meas/extensions/ngmix/LMSimpleShape.py

```diff
@@ -0,0 +1,238 @@
+    #   This metadata will be fetched from Erin when he adds an API for it
+    def getModelInfo(self, model):
+        modelInfo = {"gauss": {"nPars": 6, "nGauss": 1},
+                     "dev": {"nPars": 6, "nGauss": 6},
+                     "exp": {"nPars": 6, "nGauss": 6}}
+        if model not in modelInfo.keys():
+            raise lsst.pex.exceptions.RuntimeError("Illegal model: %s for LMSimple"%model)
+        return modelInfo[model]
+
+    def __init__(self, config, name, schema, metadata):
+        SingleFramePlugin.__init__(self, config, name, schema, metadata)
+
+        # Add ShapeletFunction keys for the number of Gaussians required
+        self.keys = []
+        self.nGauss = self.getModelInfo(self.config.model)["nGauss"]
+        #   Add a 0th order shapelet for every Gaussian to be returned
+        for i in range(self.nGauss):
+            key = ShapeletFunctionKey.addFields(schema, "%s_%d"%(name, i), "ngmix Shape gaussian",
+                                                "pixels", "", 0, HERMITE)
+            self.keys.append(key)
+        #   If the psfName is specified, it means to we can get the psfApprox from it
+        self.msfKey = MultiShapeletFunctionKey(self.keys)
+        if not self.config.psfName is None:
+            self.psfMsfKey = MultiShapeletFunctionKey(schema[self.config.psfName], HERMITE)
+        else:
+            self.psfMsfKey = None
+        self.name = name
+
+    def measure(self, measRecord, exposure):
+        psf = exposure.getPsf()
+        if psf is None:
+            raise MeasurementError(self.flagHandler.getDefinition(
+                                   SingleFrameNgmixPlugin.ErrEnum.flag_noPsf).doc,
+                                   SingleFrameNgmixPlugin.ErrEnum.flag_noPsf)
+        # make an observation for the psf image
+        psfArray = psf.computeKernelImage().getArray()
+        psfJacob = ngmix.UnitJacobian(row=(psfArray.shape[0] - 1)/2, col=(psfArray.shape[1] - 1)/2)
+        psfObs = Observation(psfArray, jacobian=psfJacob)
+
+        #   Fallback code if no psf algorithm is requested.  Then just do an LM single gaussian fit
+        if self.config.psfName is None:
+            pfitter = LMSimple(psfObs, 'gauss')
+            #   gues parameters for a simple Gaussian
+            psfPars = [0.0, 0.0, -0.03, 0.02, 8.0, 1.0]
+            pfitter.go(psfPars)
+            psfGMixFit = pfitter.get_gmix()
+            psfObs.set_gmix(psfGMixFit)
+        else:
+            shape = psf.computeImage().getArray().shape
+            image = lsst.afw.image.ImageD(shape[1], shape[0])
+            eval = measRecord.get(self.psfMsfKey).evaluate()
+            eval.addToImage(image.getArray(), lsst.afw.geom.Point2I(-(shape[0] - 1)/2, -(shape[1] - 1)/2))
+            psfObs = Observation(image.getArray(), jacobian=psfJacob)
+            #   Now create a gmix directly from what the PsfApprox algorithm produced
+            msf = measRecord.get(self.psfMsfKey)
```

Comment:
I am putting this comment here, but it applies to code above and below as well. Do not use small variable names like this. As someone who edits the code in the future this is completely opaque. Change variable names to be descriptive of what they hold or what they do